plugins/slack_app/app/controllers/slack_app_controller.rb

```diff
@@ -0,0 +1,152 @@
+require 'slack'
+
+class SlackAppController < ApplicationController
+  skip_before_action :verify_authenticity_token, only: [:command, :interact]
+  skip_around_action :login_user, except: :oauth
+
+  before_action do
+    if params[:ssl_check]
+      render json: 'ok'
+    end
+
+    @payload = params[:payload] ? JSON.parse(params[:payload]) : params
```

Comment:
Weirdness in the Slack API. The `command` endpoint is called with the fields we want right in the parameters, but the `interact` endpoint has a double-stringified JSON payload in the `payload` field of the main body. Not sure why they did this, but I didn't make the world.